src/main/kotlin/uk/gov/dluhc/printapi/database/repository/PrintDetailsRepository.kt

```diff
@@ -29,6 +33,15 @@ class PrintDetailsRepository(client: DynamoDbEnhancedClient, tableConfig: Dynamo
         }
     }
 
+    fun getAllByStatusAndBatchId(status: Status, batchId: String): List<PrintDetails> {
```

Comment:
Oh my days! I have so much to learn about dynamo! 😬 
(that says more about me than your code here, which I'm sure is absolutely fine 😁 )

Reply:
Possibly will need to revisit this once we have a larger data set to verify against.
Working with Dynamo there is much closer coupling between the database indexes and how the query is stated - not like with MySQL where we just are for data, here we also say what index to use :-)